dev/com.ibm.ws.microprofile.metrics.3.0.monitor/src/com/ibm/ws/microprofile/metrics/monitor/MetricsJaxRsEMCallbackImpl.java

```diff
@@ -0,0 +1,98 @@
+import com.ibm.ws.runtime.metadata.ComponentMetaData;
+import com.ibm.ws.threadContext.ComponentMetaDataAccessorImpl;
+
+import java.lang.reflect.Method;
+import java.util.AbstractMap;
+import java.util.Collections;
+import java.util.Map;
+
+import javax.ws.rs.container.ResourceInfo;
+
+import org.eclipse.microprofile.metrics.Counter;
+import org.eclipse.microprofile.metrics.Metadata;
+import org.eclipse.microprofile.metrics.MetricID;
+import org.eclipse.microprofile.metrics.MetricRegistry;
+import org.eclipse.microprofile.metrics.MetricType;
+import org.eclipse.microprofile.metrics.Tag;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.component.annotations.ConfigurationPolicy;
+
+@Component(service = {
+		DefaultExceptionMapperCallback.class }, configurationPolicy = ConfigurationPolicy.IGNORE, property = {
+				"service.vendor=IBM" })
+public class MetricsJaxRsEMCallbackImpl  implements DefaultExceptionMapperCallback {
+
+	
+	public static final String EXCEPTION_KEY = MetricsJaxRsEMCallbackImpl.class.getName() + ".Exception";
+	
+	
+	public synchronized static Counter registerOrRetrieveRESTUnmappedExceptionMetric(String fullyQualifiedClassName, String methodSignature) {
+		MetricRegistry baseMetricRegistry = sharedMetricRegistry.getOrCreate(MetricRegistry.Type.BASE.getName());
+		
+		Metadata metadata = Metadata.builder().withType(MetricType.COUNTER).withName("REST.request.unmappedException.total").withDescription("REST.request.unmappedException.description").withDisplayName("Total Unmapped Exception Requests").build();
+		
+		Tag classTag = new Tag("class", fullyQualifiedClassName);
+		Tag methodTag = new Tag("method", methodSignature);
+		
+		Counter counter = baseMetricRegistry.counter(metadata, classTag, methodTag);
+		
+		ComponentMetaData cmd = ComponentMetaDataAccessorImpl.getComponentMetaDataAccessor().getComponentMetaData();
+		
+		sharedMetricRegistry.associateMetricIDToApplication(new MetricID(metadata.getName(), classTag, methodTag), cmd.getJ2EEName().getApplication(), baseMetricRegistry);
+		
+		return counter;
+	}
+	
+	@Override
+	public Map<String, Object> onDefaultMappedException(Throwable t, int statusCode, ResourceInfo resourceInfo) {
+		Map.Entry<String, String> classXmethod = resolveSimpleTimerClassMethodTags(resourceInfo);
+
+		registerOrRetrieveRESTUnmappedExceptionMetric(classXmethod.getKey() ,classXmethod.getValue()).inc();
+		
+		return Collections.singletonMap(EXCEPTION_KEY, t);
+	}
+
+	static SharedMetricRegistries sharedMetricRegistry;
+	
+	@Reference
+    public void getSharedMetricRegistries(SharedMetricRegistries sharedMetricRegistry) {
+		MetricsJaxRsEMCallbackImpl.sharedMetricRegistry = sharedMetricRegistry;
+    }
+	
+	
+	private Map.Entry<String, String> resolveSimpleTimerClassMethodTags(ResourceInfo resourceInfo) {
+		Class<?> resourceClass = resourceInfo.getResourceClass();
```

Comment:
Do you need a null check on `resourceInfo`?

Reply:
Hm, this is called by the callback handler in response to a JAX-RS request that ended with an unmapped exception. I don't think the `ResourceInfo` would be `null`. @jim-krueger  what do you think?

Reply:
Correct @Channyboy , resourceInfo should never be null.